src/com/phonebookservice/server/FileDataAccessAdapter.java

```diff
@@ -53,4 +98,52 @@ public Contact update(final Long contactId, final Contact contact) {
     public void delete(final Long contactId) {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * save contacts.
+     *
+     * @throws IOException
+     */
+    @Override
+    public void save() {
+        // TODO to be done every time duration
+        try {
+            final Writer fileWriter = new FileWriter(FILE_NAME, false);
+
+            if (CollectionUtility.isNotNullOrEmptyList(this.myContactList)) {
+                for (Contact contact : this.myContactList) {
+                    fileWriter.write(ContactConverter
+                            .mapContactToString(this.myContactMap, contact));
+                }
+            }
+
+            fileWriter.close();
```

Comment:
It is better to add write and read of the file in a separate class so it can be generic for any database if want to read and write.
So what is needed to be done is : 
create an interface and create a class for file to add 2 methods (read &write)